### PR TITLE
yaml-cpp: 0.7.0 -> 0.8.0

### PR DESCRIPTION
**Pull request by tobim**
## Description of changes

https://github.com/jbeder/yaml-cpp/releases/tag/0.8.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/yaml-cpp/default.nix
Comment (OPNA2608) [inline pkgs/development/libraries/yaml-cpp/default.nix:15]: Please also drop the `rev-prefix = "yaml-cpp-"` down in `passthru.updateScript`.

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by OPNA2608** [comment]
Breaks `opencolorio`, which in turn breaks `blender`, `olive-editor`, `krita` etc:
```
CMake Error at share/cmake/modules/Findyaml-cpp.cmake:37 (get_target_property):
  get_target_property() called with non-existent target "yaml-cpp".
Call Stack (most recent call first):
  share/cmake/modules/FindExtPackages.cmake:33 (find_package)
  CMakeLists.txt:261 (include)


CMake Error at /nix/store/p6nyqsw5qbg114gprcj4ci8j4nx9nhnb-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find yaml-cpp (missing: yaml-cpp_LIBRARY) (found suitable version
  "0.8.0", minimum required is "0.7.0")
Call Stack (most recent call first):
  /nix/store/p6nyqsw5qbg114gprcj4ci8j4nx9nhnb-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  share/cmake/modules/Findyaml-cpp.cmake:114 (find_package_handle_standard_args)
  share/cmake/modules/FindExtPackages.cmake:33 (find_package)
  CMakeLists.txt:261 (include)
```

**Comment by tobim** [comment]
I think we should wait for https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1823 then.

**Comment by tobim** [comment]
krita, osl, and olive-editor are now fixed.
Mir and sysdig seem to be unrelated build failures.
mivisionx and xfitter don't get built because of other broken dependencies.

**Comment by OPNA2608** [comment]
Mir: Failing on Hydra after recent staging merge.
```
[ RUN ] RealKMSOutputTest.drm_set_gamma
/build/source/tests/unit-tests/platforms/gbm-kms/kms/test_real_kms_output.cpp:503: Failure
Actual function call count doesn't match EXPECT_CALL(mock_drm, drmModeCrtcSetGamma(drm_fd, crtc_ids[0], gamma.red.size(), const_cast<uint16_t*>(gamma.red.data()), const_cast<uint16_t*>(gamma.green.data()), const_cast<uint16_t*>(gamma.blue.data())))...
Expected: to be called once
Actual: never called - unsatisfied and active
[ FAILED ] RealKMSOutputTest.drm_set_gamma (0 ms)
```
Looks to be https://github.com/MirServer/mir/issues/3019, so unrelated.

sysdig on hardened kernels: Failing since *checks date* 2022-04-17 according to Hydra, wow. *Very likely* unrelated.

I'll run a nixpkgs-review tonight to verify.

**Comment by tobim** [comment]
`minivisonx` fails because `rapidjson-unstable` doesn't pass it's valgrind test. The attributes that are modified here are not in that closure.
`xfitter` fails because `root` seems to have a broken patch. That should also not be caused by this PR.

---
Result of `nixpkgs-review pr 249947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>redpanda-server</li>
  </ul>
</details>
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4_hardened.sysdig</li>
    <li>mir</li>
    <li>mir.dev</li>
    <li>mir.doc</li>
    <li>miriway</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu</li>
    <li>rocmPackages.mivisionx-opencl</li>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>83 packages built:</summary>
  <ul>
    <li>ashuffle</li>
    <li>blender</li>
    <li>blender-hip</li>
    <li>brise</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>contour</li>
    <li>contour.terminfo</li>
    <li>coreboot-configurator</li>
    <li>d-seams</li>
    <li>dcgm</li>
    <li>ebpf-verifier</li>
    <li>emacsPackages.rime</li>
    <li>embree</li>
    <li>embree2</li>
    <li>facter</li>
    <li>fcitx5-rime</li>
    <li>huggle</li>
    <li>ibus-engines.rime</li>
    <li>interception-tools</li>
    <li>ip2unix</li>
    <li>jami</li>
    <li>krita</li>
    <li>librime</li>
    <li>linuxKernel.packages.linux_4_19.sysdig</li>
    <li>linuxKernel.packages.linux_5_10.sysdig</li>
    <li>linuxKernel.packages.linux_5_10_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_15.sysdig</li>
    <li>linuxKernel.packages.linux_5_15_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4.sysdig</li>
    <li>linuxKernel.packages.linux_6_1.sysdig</li>
    <li>linuxKernel.packages.linux_hardened.sysdig (linuxKernel.packages.linux_6_1_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_6_5.sysdig</li>
    <li>linuxKernel.packages.linux_6_5_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_latest_libre.sysdig</li>
    <li>linuxKernel.packages.linux_libre.sysdig</li>
    <li>linuxKernel.packages.linux_lqx.sysdig</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod_latest.sysdig (linuxKernel.packages.linux_xanmod_stable.sysdig)</li>
    <li>linuxKernel.packages.linux_zen.sysdig</li>
    <li>lomiri.deviceinfo</li>
    <li>lomiri.deviceinfo.bin</li>
    <li>lomiri.deviceinfo.dev</li>
    <li>luaformatter</li>
    <li>micromamba</li>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-4_4</li>
    <li>mongodb-5_0</li>
    <li>nnpdf (python311Packages.nnpdf)</li>
    <li>olive-editor</li>
    <li>opencolorio</li>
    <li>openimageio</li>
    <li>openimageio.bin</li>
    <li>openimageio.dev</li>
    <li>openimageio.doc</li>
    <li>openloco</li>
    <li>openmw</li>
    <li>openmw-tes3mp</li>
    <li>openxcom</li>
    <li>osl</li>
    <li>pdns</li>
    <li>portmod</li>
    <li>portmod.dist</li>
    <li>prometheus-dcgm-exporter</li>
    <li>python310Packages.nnpdf</li>
    <li>python310Packages.validphys2</li>
    <li>python310Packages.validphys2.dist</li>
    <li>validphys2 (python311Packages.validphys2)</li>
    <li>validphys2.dist (python311Packages.validphys2.dist)</li>
    <li>qdmr</li>
    <li>qtcreator</li>
    <li>rime-data</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>sysdig</li>
    <li>thinkfan</li>
    <li>timeloop</li>
    <li>torrenttools</li>
    <li>way-displays</li>
    <li>yaml-cpp</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
`aarch64-darwin` OfBorg says `opencolorio-2.3.0` fails because it tries to fetch [DLTcollab/sse2neon](https://github.com/DLTcollab/sse2neon) via [CMake `FetchContent`](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/8add374dade5a7e3ee802269b71e1c0b75f8b8f0/share/cmake/modules/install/Installsse2neon.cmake#L14-L20). Either fetch & link it in a way that CMake is happy with it or disable it with [`-DOCIO_USE_SSE2NEON=OFF`](https://github.com/AcademySoftwareFoundation/OpenColorIO/commit/caa20dd1962f0d659a9214ac28815a1b904d4118#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR270-R274).

**Comment by tobim** [comment]
Result of `nixpkgs-review pr 249947` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages marked as broken and skipped:</summary>
  <ul>
    <li>contour</li>
    <li>contour.terminfo</li>
    <li>mongodb-4_4</li>
    <li>mongodb-5_0</li>
    <li>nnpdf</li>
    <li>openmw</li>
    <li>python310Packages.nnpdf</li>
    <li>python310Packages.validphys2</li>
    <li>python310Packages.validphys2.dist</li>
    <li>python311Packages.nnpdf</li>
    <li>python311Packages.validphys2</li>
    <li>python311Packages.validphys2.dist</li>
    <li>torrenttools</li>
    <li>validphys2</li>
    <li>validphys2.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>embree</li>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>ashuffle</li>
    <li>luaformatter</li>
    <li>micromamba</li>
    <li>mongodb (mongodb-6_0)</li>
    <li>opencolorio</li>
    <li>openimageio</li>
    <li>openimageio.bin</li>
    <li>openimageio.dev</li>
    <li>openimageio.doc</li>
    <li>osl</li>
    <li>timeloop</li>
    <li>yaml-cpp</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
Sorry for the review delay, should have more free time for this now. Re-running the review to test everything.

Regarding the changes:
- The ones to `yaml-cpp` look fine to me, as a maintainer of that package.
- The patches to `krita` and `olive-editor` & fix to `openimageio` seem minor enough to be okay in principle.
  - In the case of the `olive-editor` one, I think it's common practice to at least add a comment linking to the open PR. If you open the commit URL, it's not possible to immediately tell what repo/branch/PR this belongs to.
  - For the `openimageio` one, a link to the upstream CMake issue would be nice, like you did for the `opencolorio` one.
- I'd prefer if one of the pinged maintainers had a look over the changes to `opencolorio`. IMO, on x86_64, it would likely be better to disable all `OCIO_USE_*` SIMD options not available in the base x86_64-v1 we're still aiming for until https://github.com/NixOS/nixpkgs/pull/202526 or another PR bumps it to something newer. According to https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels, of the available SIMD options, v1 should only guarantee SSE2.

**Comment by tobim** [comment]
> IMO, on x86_64, it would likely be better to disable all OCIO_USE_* SIMD options not available in the base x86_64-v1 we're still aiming for until systems/architecture: bump default architecture to x86-64-v2  #202526 or another PR bumps it to something newer. According to https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels, of the available SIMD options, v1 should only guarantee SSE2.

OpenColorIO checks which extensions are supported by the CPU at runtime, so it should actually be fine to enable all of them. The interface for that is defined in https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/src/OpenColorIO/CPUInfo.h.

**Comment by OPNA2608** [comment]
> OpenColorIO checks which extensions are supported by the CPU at runtime, so it should actually be fine to enable all of them.

Was there a reason why you disabled all the AVX ones then? Or just uncertainty about how this worked?

I guess for the `SSE2NEON` one, a comment about it requiring another download & `FetchContent` handling would also be good, to explain why we're turning only that one off. If you *could* still get it working that be great, but it's new functionality so disabling it for now seems fine to me.

(`nixpkgs-review` still running, has some large - size and time - rebuilds)

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 249947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>redpanda-server</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_19_hardened.sysdig (linuxKernel.packages.linux_4_19_hardened.sysdig)</li>
    <li>linuxPackages_5_4_hardened.sysdig (linuxKernel.packages.linux_5_4_hardened.sysdig)</li>
    <li>linuxPackages_latest.sysdig (linuxKernel.packages.linux_6_6.sysdig)</li>
    <li>linuxPackages_latest-libre.sysdig (linuxKernel.packages.linux_latest_libre.sysdig)</li>
    <li>linuxPackages_zen.sysdig (linuxKernel.packages.linux_zen.sysdig)</li>
    <li>mir</li>
    <li>mir.dev</li>
    <li>mir.doc</li>
    <li>miriway</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip ,rocmPackages_5.mivisionx ,rocmPackages_5.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_5.mivisionx-cpu)</li>
    <li>rocmPackages.mivisionx-opencl (rocmPackages_5.mivisionx-opencl)</li>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>86 packages built:</summary>
  <ul>
    <li>ashuffle</li>
    <li>blender</li>
    <li>blender-hip</li>
    <li>bloom</li>
    <li>brise</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>contour</li>
    <li>contour.terminfo</li>
    <li>coreboot-configurator</li>
    <li>d-seams</li>
    <li>dcgm</li>
    <li>ebpf-verifier</li>
    <li>emacsPackages.rime</li>
    <li>embree</li>
    <li>embree2</li>
    <li>facter</li>
    <li>fcitx5-rime</li>
    <li>huggle</li>
    <li>ibus-engines.rime</li>
    <li>interception-tools</li>
    <li>ip2unix</li>
    <li>jami</li>
    <li>krita</li>
    <li>librime</li>
    <li>linuxKernel.packages.linux_4_19.sysdig</li>
    <li>linuxKernel.packages.linux_5_10.sysdig</li>
    <li>linuxPackages_5_10_hardened.sysdig (linuxKernel.packages.linux_5_10_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_5_15.sysdig</li>
    <li>linuxPackages_5_15_hardened.sysdig (linuxKernel.packages.linux_5_15_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_5_4.sysdig</li>
    <li>linuxPackages.sysdig (linuxKernel.packages.linux_6_1.sysdig)</li>
    <li>linuxPackages_hardened.sysdig (linuxPackages_6_1_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_6_5.sysdig</li>
    <li>linuxPackages_6_5_hardened.sysdig (linuxKernel.packages.linux_6_5_hardened.sysdig)</li>
    <li>linuxPackages-libre.sysdig (linuxKernel.packages.linux_libre.sysdig)</li>
    <li>linuxPackages_lqx.sysdig (linuxKernel.packages.linux_lqx.sysdig)</li>
    <li>linuxPackages_testing_bcachefs.sysdig (linuxKernel.packages.linux_testing_bcachefs.sysdig)</li>
    <li>linuxPackages_xanmod.sysdig (linuxKernel.packages.linux_xanmod.sysdig)</li>
    <li>linuxPackages_xanmod_latest.sysdig (linuxKernel.packages.linux_xanmod_latest.sysdig ,linuxPackages_xanmod_stable.sysdig)</li>
    <li>lomiri.deviceinfo</li>
    <li>lomiri.deviceinfo.bin</li>
    <li>lomiri.deviceinfo.dev</li>
    <li>luaformatter</li>
    <li>micromamba</li>
    <li>mongodb (mongodb-6_0)</li>
    <li>mongodb-4_4</li>
    <li>mongodb-5_0</li>
    <li>nnpdf (python311Packages.nnpdf)</li>
    <li>olive-editor</li>
    <li>opencolorio</li>
    <li>openimageio</li>
    <li>openimageio.bin</li>
    <li>openimageio.dev</li>
    <li>openimageio.doc</li>
    <li>openloco</li>
    <li>openmw</li>
    <li>openmw-tes3mp</li>
    <li>openxcom</li>
    <li>osl</li>
    <li>pdns</li>
    <li>portmod</li>
    <li>portmod.dist</li>
    <li>prometheus-dcgm-exporter</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.nnpdf</li>
    <li>python310Packages.validphys2</li>
    <li>python310Packages.validphys2.dist</li>
    <li>python311Packages.n3fit</li>
    <li>python311Packages.n3fit.dist</li>
    <li>validphys2 (python311Packages.validphys2)</li>
    <li>validphys2.dist (python311Packages.validphys2.dist)</li>
    <li>qdmr</li>
    <li>qtcreator</li>
    <li>rime-data</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>sysdig</li>
    <li>thinkfan</li>
    <li>timeloop</li>
    <li>torrenttools</li>
    <li>way-displays</li>
    <li>yaml-cpp</li>
  </ul>
</details>

---

Checked failure reasons, saw nothing that would be caused by this PR as you mentioned. Checked `mir` and `miriway` by applying #255029 on top to fix the tests, also works. Aside from the question of the SIMD flags, LGTM.

**Comment by tobim** [comment]
> Is there a reason why we disabled all the AVX ones then? Or just uncertainty about how this worked?

Frankly I forgot to take it out again. I only looked in detail after setting the flags. I can remove them in a few hours.

**Review by OPNA2608 — APPROVED** [review]
Re-checked `opencolorio` cus that's the only change since last `nixpkgs-review` (already GC'd that one, and I don't wanna sit through all the big rebuilds again…), still builds. LGTM?

Thanks for working on this basically as soon as the new version released, much appreciated!

File: pkgs/applications/video/olive-editor/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/video/olive-editor/default.nix:38]: that's good 👍🏼 we should never overwrite patchPhase

**Pull request merged**
